src/lib/tools/session.ts

```diff
@@ -0,0 +1,44 @@
+import { jwtVerify, SignJWT } from "jose";
+import { randomBytes } from "crypto";
+import { NextResponse } from "next/server";
+
+// Environment variable for the secret key
+const secretKey = process.env.SESSION_SECRET || randomBytes(64).toString("hex");
```

Comment:
We shouldn't try to randomly generate a secret here as that would break sessions if misused. Instead we should just throw an error to let us know we need to store a secret.

Reply:
Noted, I'll refactor to fit this change